src/views/instances/new-dhis2/fields/image-tag-select.tsx

```diff
@@ -63,19 +45,80 @@ export const ImageTagSelect: FC<{ displayName: string }> = ({ displayName }) =>
         }
     }, [repository, refetch])
 
-    const optionsWithFallback = options.length === 0 && initialValue ? [{ value: initialValue, label: initialValue }] : options
+    useEffect(() => {
+        if (data) {
+            const mappedOptions = data.map(mapStringToValueLabel)
+            setOptions(mappedOptions)
+
+            const currentSelectedValue = form.getState().values['dhis2-core']?.IMAGE_TAG
+            if (currentSelectedValue && !mappedOptions.some((option) => option.value === currentSelectedValue)) {
+                form.change(IMAGE_TAG_FIELD_NAME, undefined)
+                form.blur(IMAGE_TAG_FIELD_NAME)
+            }
+        }
+    }, [data, form])
+
+    const handleTagCheck = useCallback(
+        async (tag: string): Promise<boolean> => {
+            const existsInOptions = options.some((option) => option.label.startsWith(tag))
+            setTagExists(existsInOptions)
+
+            if (existsInOptions) {
+                setOptions((prevOptions) => {
+                    const filteredOptions = prevOptions.filter((option) => option.value !== tag)
+                    return [{ value: tag, label: tag }, ...filteredOptions]
+                })
+                return true
+            }
```

Comment:
This could be optimized a little more:

```suggestion
            const filteredExistingOptions = options.filter((option) => option.value.startsWith(tag))
            const existsInOptions = filteredExistingOptions.length
            setTagExists(existsInOptions)

            if (existsInOptions) {
                setOptions(filteredExistingOptions)
                return
            }
```

This way the user will get all the options that start with the input, everything else will be hidden.
I think it's safe to assume that the user doesn't care about versions he didn't look for.

Reply:
I have to add `{ value: tag, label: tag }` to the options as the select component always checks to see if what was typed matches anything in the list.

The filteredOptions is cleanup to prevent duplicate tags from being in the option.

Writing the suggested code may not account for this, which causes errors.

Reply:
Thanks for the approval